common/djangoapps/nodebb/tasks.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from celery.exceptions import MaxRetriesExceededError
+from celery.task import task  # pylint: disable=no-name-in-module, import-error
+
+from common.lib.nodebb_client.client import NodeBBClient
+from django.conf import settings
+
+log = logging.getLogger('edx.celery.task')
+
+
+@task(bind=True, default_retry_delay=settings.NODEBB_TASK_DELAY, max_retries=settings.NODEBB_TASK_MAX_TRIES)
+def create_category_on_nodebb_task(self, course_name):
+    status_code, response_body = NodeBBClient().categories.create(name=course_name)
+
+    if status_code != 200:
+        try:
+            self.retry()
+        except MaxRetriesExceededError:
+            log.error(
+                "Error: Can't create nodebb community for the given course {} due to {}".format(
+                    course_name, response_body
+                )
+            )
+    else:
+        log.info('Success: Community created for course {}'.format(course_name))
+
+
+@task(bind=True, default_retry_delay=settings.NODEBB_TASK_DELAY, max_retries=settings.NODEBB_TASK_MAX_TRIES)
```

Comment:
why we need `bind=True` here?

Reply:
A [task being bound](http://docs.celeryproject.org/en/latest/userguide/tasks.html) means the first argument to the task will always be the task instance (self), just like Python bound methods.
**We are using that to retry a task**